src/pgwire/src/protocol.rs

```diff
@@ -277,6 +293,7 @@ where
     let machine = StateMachine {
         conn,
         adapter_client,
+        update_claims_fn,
```

Comment:
It seems funny to me to bake this into the pgwire state machine. I'd love to instead make it part of the _session_. It seems analogous to the `notices_tx` and `notices_rx` that we currently stash on the session.

Proposal:

  * Add `external_metadata_tx` and `external_metadata_rx` to the session.
  * Add a `retain_external_metadata_transmitter()` method to the session.
  * Add an `apply_external_metadata_updates` method to the session, which drains any external metadata updates and applies the changes from the latest.
  * Have `continuously_validate_access_token` take _as input_ a function which should be invoked whenever the claims are updated.
  * Have the pgwire protocol module pass in a function that converts the claims to external user metadata and then sends them to `external_metadata_tx`. (The closure should hold on to `Session::retain_external_metadata_transmitter()`.)
  * Have `Coord::handle_message` call `session.apply_external_metadata_updates` before invoking any command.

I think you can even simplify the signature of `continuously_validate_access_token` to _not_ return claims, since the first set of claims will get delivered over the channel immediately and will be available to the coordinator.

This will make it easier to rig up this logic to the HTTP server, if we decide that e.g. we want to do continuous validation there, too, for continuous `SUBSCRIBE` queries or somesuch.

Reply:
I pushed a first pass attempt at this. I think I got it right, but I'm going to look it over again tomorrow.

Reply:
Yeah I think it looks good. I'm still a bit torn about whether the initial claims should be explicitly returned or processed by the passed in closure. Do you have any thoughts after seeing it?